WebRoot/js/ajax/dwt/core/Dwt.js

```diff
@@ -556,7 +556,13 @@ function(htmlElement, point) {
 Dwt.setLocation =
 function(htmlElement, x, y) {
 	if (!(htmlElement = Dwt.getElement(htmlElement))) { return; }
-	var position = DwtCssStyle.getProperty(htmlElement, 'position');
+	var position;
+	if (AjxEnv.isMSEdge && AjxEnv.browserVersion > 18) {
+		position = htmlElement.style.position;
+	}
+	if (!position) {
+		position = DwtCssStyle.getProperty(htmlElement, 'position');
```

Comment:
one more thought
if you see code in DwtCssStyle.getProperty then it is either trying to access htmlElement.currentStyle.position or htmlElement.style.position, so should we move this fix to DwtCssStyle.getProperty ?
also I am more interested to know why DwtCssStyle.getProperty returns undefined in edge 44, what was exactly changed in that version

Reply:
First, I correct my comment.
(correct) `DwtCssStyle.getProperty(htmlElement, 'position')` returns **an empty string** on Edge44.

Regarding the second comment, the function `getComputedStyle` is called in DwtCssStyle.getProperty.
https://github.com/Zimbra/zm-ajax/blob/8.8.11/WebRoot/js/ajax/dwt/graphics/DwtCssStyle.js line 115.
(reference) https://msdn.microsoft.com/en-us/ie/ff975168(v=vs.94)

On Edge 42, `cssDecl`, which is the returned value (object) of getComputedStyle, has values of styles. Then `return cssDecl.getPropertyValue(cssPropName);` returns "absolute".
![edge42](https://user-images.githubusercontent.com/32184593/52629623-779ea500-2efd-11e9-8216-fcd0214b8f67.png)

On Edge 44, `cssDecl` doesn't have values of styles in the following cases (callstacks) at loading page.
(1)
Dwt.setLocation
Dwt.setBounds
DwtShell.prototype._createBusyOverlay
DwtShell
ZmZimbraMail.run
launch

(2)
Dwt.setLocation
Dwt.setBounds
DwtShell
ZmZimbraMail.run
launch
![edge44](https://user-images.githubusercontent.com/32184593/52629655-8be2a200-2efd-11e9-9644-30aa805cb9d1.png)

All styles are empty. Then `return cssDecl.getPropertyValue(cssPropName);` returns an empty string.
The behavior of `getComputedStyle` seems to be changed on Edge 44.

Regarding the first comment, `DwtCssStyle.getProperty` is called not only from Dwt.setLocation but also from other functions.
i.e. it is used not only for "position" but also for "top", "display" etc.
In my opinion, we should not implement a logic for "position" in `DwtCssStyle.getProperty` but keep it as a common function for any styles.

Reply:
Actual issue was simple, in Edge 44, getComputedStyle returned empty values because when you create new element and try to call getComputedStyle on that element before putting it inside DOM hierarchy then browser will not be able to compute styles from DOM

I have fixed it in a elegant manner in https://github.com/Zimbra/zm-ajax/pull/75